### PR TITLE
APPS/IO-DEMO: Fix reporting performance when no data completed

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Fix reporting performance when no data completed on IO-Demo Client/Server.

## Why ?

Fixes:
```
[1625154064.946534] [DEMO] read 0.801156 MBs min:0(2.1.3.8:57414) max:2 total:35 | write 0.564616 MBs min:0(2.1.3.8:57414) max:0 total:9 | active: 101/101, buffers:690
```
i.e. reporting that total completed - `9`, but maximum completed per servers - `0`.

## How ?

1. Fix `update_min_max()` to correctly calculate `min`/`max` using number of completions instead of bytes. Bytes can be written, when not all fragments are received, but completion counter isn't incremented in case of error.
2. Fix `update_min_max()` to correctly calculate `max` if some iterator is `min` and `max` by removing `else` for `max`.
3. Fix `report_performance()` on a Client by removing `op_info_t` (no longer needed) and calculate `total_compelted` inside `report_performance()` from all completed operations. Since the operation could be sent, but not completed yet. Relying on previous `op_info_t::num_iters` from `run()` loop was incorrect - it is not adjusted when operation isn't really sent due to error.

### CODE REVIEW

**Comment by dmitrygx** [comment]
@hoopoepg @brminich @evgeny-leksikov could you review pls?

**Review by hoopoepg — APPROVED** [review]

**Review by evgeny-leksikov — APPROVED** [review]
👍 just couple of minor comments

**Review by brminich — APPROVED** [review]

File: test/apps/iodemo/io_demo.cc [hunk 1]
```diff
@@ -1926,7 +1922,8 @@ class DemoClient : public P2pDemoCommon {
                                  size_t(0), opts().operations.size() - 1)];
     }
 
-    void report_performance(long num_iters, double elapsed, op_info_t *op_info) {
+    void report_performance(long num_iters, double elapsed,
+                            size_t *total_bytes) {
```
[the remark this answers is not in the document]
Comment (dmitrygx) [reply test/apps/iodemo/io_demo.cc:1926]: done, but can't be `const`

File: test/apps/iodemo/io_demo.cc [hunk 2]
```diff
@@ -1947,14 +1944,14 @@ class DemoClient : public P2pDemoCommon {
             first_print = false;
 
             // Report bandwidth
-            double throughput_mbs = op_info[op_id].total_bytes /
-                                    elapsed / (1024.0 * 1024.0);
+            double throughput_mbs = total_bytes[op_id] / elapsed /
+                                    (1024.0 * 1024.0);
```
[the remark this answers is not in the document]
Comment (dmitrygx) [reply test/apps/iodemo/io_demo.cc:1948]: done

**Review by evgeny-leksikov — APPROVED** [review]

**Comment by dmitrygx** [comment]
@yosefe could you review pls?

**Review by yosefe — APPROVED** [review]

**Pull request merged**